ui/tx/interpretation/TxInterpretation.tsx

```diff
@@ -0,0 +1,94 @@
+import { Skeleton, Flex, Text, Icon, chakra } from '@chakra-ui/react';
+import BigNumber from 'bignumber.js';
+import React from 'react';
+
+import type { TxInterpretationSummary, TxInterpretationVariable } from 'types/api/txInterpretation';
+
+import config from 'configs/app';
+import actionIcon from 'icons/action.svg';
+import dayjs from 'lib/date/dayjs';
+import AddressEntity from 'ui/shared/entities/address/AddressEntity';
+import TokenEntity from 'ui/shared/entities/token/TokenEntity';
+
+import { extractVariables, getStringChunks, NATIVE_COIN_SYMBOL_VAR_NAME } from './utils';
+
+type Props = {
+  summary?: TxInterpretationSummary;
+  isLoading?: boolean;
+  className?: string;
+}
+
+const TxInterpretationElementByType = ({ variable }: { variable?: TxInterpretationVariable }) => {
+  if (!variable) {
+    return null;
+  }
+
+  const { type, value } = variable;
+  switch (type) {
+    case 'address':
+      return <AddressEntity address={ value } truncation="constant" sx={{ ':not(:first-child)': { marginLeft: 1 } }} whiteSpace="initial"/>;
+    case 'token':
+      return (
+        <TokenEntity
+          token={ value }
+          onlySymbol
+          noCopy
+          width="fit-content"
+          sx={{ ':not(:first-child)': { marginLeft: 1 } }}
+          mr={ 2 }
+          whiteSpace="initial"
+        />
+      );
+    case 'currency': {
+      let numberString = '';
+      if (BigNumber(value).isLessThan(0.1)) {
+        numberString = BigNumber(value).toPrecision(2);
+      } else if (BigNumber(value).isLessThan(10000)) {
+        numberString = BigNumber(value).dp(2).toFormat();
+      } else {
+        numberString = BigNumber(value).dividedBy(1000).toFormat(2) + 'K';
```

Comment:
If the value is greater than a million, should we compact it with the "M" suffix then?